integrations/transportation/trainline.js

```diff
@@ -0,0 +1,149 @@
+import { get } from 'lodash';
+
+import {
+  ACTIVITY_TYPE_TRANSPORTATION,
+  TRANSPORTATION_MODE_TRAIN,
+  TRANSPORTATION_MODE_BUS,
+  TRANSPORTATION_MODE_PUBLIC_TRANSPORT,
+} from '../../definitions';
+import { ValidationError, AuthenticationError } from '../utils/errors';
+import parseCookies from '../authentication/parseCookies';
+import request from '../utils/request';
+
+const LOGIN_PATH = 'https://www.thetrainline.com/login-service/api/login';
+const PAST_BOOKINGS_PATH = 'https://www.thetrainline.com/my-account/api/bookings/past';
+
+function matchTransportMode(modeFromTrainline) {
+  switch (modeFromTrainline) {
+    case 'train':
+      return TRANSPORTATION_MODE_TRAIN;
+    case 'bus':
+      return TRANSPORTATION_MODE_BUS;
+    default:
+      return TRANSPORTATION_MODE_PUBLIC_TRANSPORT;
+  }
+}
+
+async function login(username, password) {
+  const loginResponse = await request(LOGIN_PATH, {
+    method: 'post',
+    data: {
+      email: username,
+      password,
+    },
+  });
+  // Check if authenticated=true on the response body
+  if (!loginResponse.data.authenticated) throw new AuthenticationError('Login failed');
+
+  // Save session cookies for authentication in subsequent requests
+  return parseCookies(loginResponse);
+}
+
+
+async function connect(requestLogin, requestWebView, logger) {
+  const { username, password } = await requestLogin();
+
+  if (!(password || '').length) {
+    throw new ValidationError('Password cannot be empty');
+  }
+  const cookies = await login(username, password);
```

Comment:
I'm worried about this as React Native doesn't play well with cookie handling.
Could you explain me what is needed here?
If it's simply to keep cookies persisted, maybe we should use a cookiejar similarly to linky.js
Unfortunately this means that the `collect()` stage can have outdated cookies and require a new `login`

Reply:
Yes, it's to store the auth cookies for next time. Will look into cookiejar